blueflood-core/src/main/java/com/rackspacecloud/blueflood/utils/GenericClassLoader.java

```diff
@@ -0,0 +1,74 @@
+
+/*
+ * Copyright 2015 Rackspace
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.rackspacecloud.blueflood.utils;
+
+import com.rackspacecloud.blueflood.service.Configuration;
+import com.rackspacecloud.blueflood.service.CoreConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Map;
+import java.util.HashMap;
+
+
+public class GenericClassLoader {
+    private static final Logger log = LoggerFactory.getLogger(GenericClassLoader.class);
+
+    private static Map<CoreConfig, Object> loadedModules = new HashMap<CoreConfig, Object>();
+
```

Comment:
I am not a fan of caching instances in this way. If we load more modules, how many instances do we cache? I think the responsibility of caching should be with the invoker. Do you agree?